sympy/solvers/inequalities.py

```diff
@@ -982,3 +986,482 @@ def reduce_inequalities(inequalities, symbols=[]):
+    """
+    A linear programing problem is considered infeasible if its constraint set
+    is empty. That is, if the set of all vectors satisfying the contraints is
+    empty, then the problem is infeasible.
+
+    Example
+    =======
+
+    Suppose you want to maximize
+        x
+    subject to
+        x >= 10
+        x <= 9
+
+    It's not possible for x to satisfy the given contraints.
+    """
+    pass
+
+
+def _pivot(M, i, j):
+    """
+    There are four rules for the pivot operation:
+    - The pivot quantity goes into its reciprocal
+    - Entries in the same row as the pivot are divided by the pivot
+    - Entries in the same column as the pivot are divided by the pivot and
+    changed in sign.
+    - The remaining entries are reduced in value by the following: the product
+    of the corresponding entries in the same row and column as themselves and
+    the pivot, divided by the pivot.
+
+    Example
+    =======
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.inequalities import _pivot
+    >>> m = Matrix([[3,1,5],
+    ...             [2,-3,1],
+    ...             [0,3,1]])
+    >>> _pivot(m, 1,0) # pivot around 2, which is in row 1 and col 0
+    Matrix([
+    [-3/2, 11/2, 7/2],
+    [ 1/2, -3/2, 1/2],
+    [   0,    3,   1]])
+    """
+    Mi, Mj, Mij = M[i,:], M[:,j], M[i,j]
+    if Mij == 0:
+        raise ZeroDivisionError("Tried to pivot about zero-valued entry.")
+    MM = M - Mj * (Mi / Mij)
+    MM[i,:] = Mi / Mij
+    MM[:,j] = -Mj / Mij
+    MM[i,j] = 1 / Mij
+    return MM
+
+
+def _choose_pivot_row(A, B, candidate_rows, pivot_col, S):
+    # Choose row with smallest ratio
+    min_ratio = float('inf')
+    min_rows = []
+    for i in candidate_rows:
+        ratio = B[i] / A[i, pivot_col]
+        if ratio < min_ratio:
+            min_ratio = ratio
+            min_rows = [i]
+        elif ratio == min_ratio:
+            min_rows.append(i)
+
+    # If there are ties, pick using Bland's rule
+    row = sorted(min_rows, key= lambda r: S[r])[0]
+    return row
+
+def _simplex(A, B, C, skip_phase_2=False):
+    """
+    Simplex method with Bland's rule
+
+    References
+    ==========
+
+    .. [1] Thomas S. Ferguson, LINEAR PROGRAMMING: A Concise Introduction
+           web.tecnico.ulisboa.pt/mcasquilho/acad/or/ftp/FergusonUCLA_LP.pdf
+    """
+    from sympy.matrices.dense import Matrix
+
+    m, n = A.shape
+    if B.shape != (m, 1) or C.shape != (1, n):
+        raise ValueError(f"The shape of matrix B ({B.shape}) or C ({C.shape})" \
+                         f"does not match the shape of matrix A ({A.shape}).")
+
+    for val in list(A) + list(B) + list(C):
+        val = sympify(val)
+        if not (val.is_rational or isinstance(val, Float)):
+            raise TypeError(f"Only rationals and floats are allowed in the Simplex method." \
+                             f" ({val}) is neither.")
+
+    D = ImmutableMatrix([0])
+    M = Matrix([[A, B], [-C, D]])
```

Comment:
```suggestion
    D = ImmutableMatrix([0])  # D is 1x1, A is mxn, B is mx1 and C is 1xn
    M = Matrix([[A, B], [-C, D]])  # error will raise if dimensions are wrong
    if not all(i.is_Float or i.is_Rational for i in M):
            raise TypeError(f"Only rationals and floats are allowed in the Simplex method." \
                             f" ({val}) is neither.")
```

Reply:
I remember reading a comment somewhere ([this comment](https://github.com/sympy/sympy/pull/21687/files#r703017372) from the old PR) that said that it would be better to raise an exception if the dimensions are wrong because it would be less confusing for the average user.

The exception should be raised in `linprog_from_matrices` though, not `_simplex`, so I'll fix that.